src/com/amazon/ion/impl/IonTypeID.java

```diff
@@ -0,0 +1,117 @@
+package com.amazon.ion.impl;
+
+import com.amazon.ion.IonType;
+
+/**
+ * Holds pre-computed information about a binary Ion type ID byte.
+ */
+final class IonTypeID {
```

Comment:
I like this class. I like the precomputation, and how succinctly it expresses validity.